curp/src/server/cmd_worker/conflict_checked_mpmc.rs

```diff
@@ -306,19 +306,18 @@ impl<C: Command> Filter<C> {
                     }
                     false
                 }
-                (ExeState::Executed(true), AsState::AfterSyncReady(index)) => {
+                (ExeState::Executed(exe_ok), AsState::AfterSyncReady(index)) => {
                     *as_st = AsState::AfterSyncing;
                     let task = Task {
                         vid,
-                        inner: Cart::new(TaskType::AS(Arc::clone(cmd), index)),
+                        inner: Cart::new(TaskType::AS(Arc::clone(cmd), index, exe_ok)),
                     };
                     if let Err(e) = self.filter_tx.send(task) {
                         error!("failed to send task through filter, {e}");
                     }
                     false
                 }
-                (ExeState::Executed(true), AsState::AfterSynced)
-                | (ExeState::Executed(false), AsState::AfterSyncReady(_)) => true,
```

Comment:
combined with the previous case, after sync will also be called after the execution fails

Reply:
I mean that a cmd that executes fail won't execute `after_synced` and it will finally remove by the `gc_spec_task`. That's the original design. But now, a cmd will execute `after_synced` whether it executes successfully or fail and a `cmd_worker` will remove this cmd from spec_pool. So, do we still need this `gc_spec_task`?

Reply:
we still need it, when the client only sends a proposal to a follower and then crash immediately, then this command needs to wait for `gc_spec_task` to delete it